vignettes/checks/sourceConceptRecordCompleteness.Rmd

```diff
@@ -23,24 +23,50 @@ The number and percent of records with a value of 0 in the source concept field
 
+### ETL Developers
+Recall that the `X_source_concept_id` columns should contain the OMOP concept representing the exact code used in the source data for a given record: “If the <x_source_value> is coded in the source data using an OMOP supported vocabulary put the concept id representing the source value here.” 
 
-### Violated rows query
-```sql
+A failure of this check usually indicates a failure to map a source value to an OMOP concept.  In some cases, such a failure can and should be remediated in the concept-mapping step of the ETL.  In other cases, it may represent a mapping that currently is not possible to implement. 
+
+To investigate the failure, run the following query: 
 
+```sql
+SELECT  
+  concept.concept_name AS standard_concept_name, 
+  cdmTable.X_concept_id, -- standard concept ID field for the table 
+  c2.concept_name AS source_value_concept_name, 
+  cdmTable.X_source_value, -- source value field for the table 
+  COUNT(*) 
+FROM @cdmDatabaseSchema.@cdmTableName cdmTable 
+LEFT JOIN @vocabDatabaseSchema.concept ON concept.concept_id = cdmTable.X_concept_id 
+-- WARNING this join may cause fanning if a source value exists in multiple vocabularies 
+LEFT JOIN @vocabDatabaseSchema.concept c2 ON concept.concept_code = cdmTable.X_source_value 
+AND c2.domain_id = <Domain of cdmTable> 
+WHERE cdmTable.@cdmFieldName = 0  
+-- AND cdmTable.value_as_number IS NOT NULL -- uncomment for unit_concept_id checks 
+GROUP BY 1,2,3 
+ORDER BY 4 DESC 
 ```
 
+The query results will give you a summary of the source codes which failed to map to an OMOP concept.  Inspecting this data should give you an initial idea of what might be going on. 
 
-### ETL Developers
+If source values return legitimate matches on concept_code, it’s possible that there is an error in the concept mapping step of your ETL.  Please note that while the `X_source_concept_id` fields are technically not required, it is highly recommended to populate them with OMOP concepts whenever possible.  This will greatly aid analysts in understanding the provenance of the data. 
 
+If source values do NOT return matches on concept_code and you are NOT handling concept mapping locally for a non-OMOP source vocabulary, then you likely have a malformed source code or one that does not exist in the OMOP vocabulary.  Please see the documentation in the [standardConceptRecordCompleteness](standardConceptRecordCompleteness.html) page for instructions on how to handle this scenario. 
 
 ### Data Users
+Since most standard OHDSI analytic workflows rely on the standard concept field and not the source concept field, failures of this check will not necessarily impact your analysis.  However, if your analysis depends on understanding source coding practices or on codes you know may not be fully mapped to OMOP standard concepts, then this will be a critical check failure to understand. 
```

Comment:
```suggestion
Since most standard OHDSI analytic workflows rely on the standard concept field and not the source concept field, failures of this check will not necessarily impact your analysis.  However, having the source concept will give you a better understanding of the provenance of the code and highlight potential issues where meaning is lost due to mapping to a standard concept.
```